core/src/main/java/org/springframework/security/oauth2/server/authorization/converter/OAuth2AuthorizationCodeAuthenticationTokenConverter.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.oauth2.server.authorization.converter;
+
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.security.oauth2.core.endpoint.OAuth2ParameterNames;
+import org.springframework.security.oauth2.server.authorization.authentication.OAuth2AuthorizationCodeAuthenticationToken;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * Converts from a {@link HttpServletRequest} to an {@link OAuth2AuthorizationCodeAuthenticationToken} that can be authenticated. The
+ * converter does not validate the request but only performs a conversion.
+ * @author Madhu Bhat
+ */
+public class OAuth2AuthorizationCodeAuthenticationTokenConverter implements Converter<HttpServletRequest, OAuth2AuthorizationCodeAuthenticationToken> {
```

Comment:
Let's make this a private method in `OAuth2TokenEndpointFilter`

Reply:
So no need of the existing `authorizationGrantConverter` converter object in `OAuth2TokenEndpointFilter` right?

Reply:
Let's keep `authorizationGrantConverter` and simply point by method reference to private method. We will likely expose a setter (later) to allow for customizations of `authorizationGrantConverter`